docs/00.MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,420 @@
+# Migration guide
+
+In this guide we gather all the information for you to update the JerryScript from 1.0 to 2.0.
+
+We have changed API functions and removed some legacy macros/headers.
+Changes are all documented below you can see them in the before and after code snippets.
+If you are looking for something that has been deleted we have a list further down.
+As for the new functions we have a list of them, for more information please checkout the [API reference](02.API-REFERENCE.md).
+
+# Modified API functions
```

Comment:
The script parsing ( #2282 ) and the source code evaluation ( #2431 ) API functions also have been modified. Please mention them in this section.

Reply:
I've added them to the guide. Thank you!